staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -65,9 +65,8 @@ type controller struct {
 	latestLease *coordinationv1.Lease
 
 	// newLeasePostProcessFunc allows customizing a lease object (e.g. setting OwnerReference)
-	// before every time the lease is created/refreshed(updated). Note that an error will block
-	// a lease CREATE, causing the controller to retry next time, but an error won't block a
-	// lease UPDATE.
+	// before every time the lease is created/refreshed(updated).
```

Comment:
This is an important change in behavior in a public library, another option can be to create a well-known error type and limit the change asserting and that specific error type, and modify the kubelet to return that error type

Reply:
Yeah, putting this code here was contentious in the original PR, because it probably *shouldn't* be a public library. I think it's strange for the interface to return an `error` but have conditions around when the error is _treated_ as an error. I can't exhaustively verify that this change doesn't break any consumers, but I have verified that it doesn't break the Kubernetes components that utilize it.